### PR TITLE
Add i18n for labels

### LINKED ISSUE #64: Add localisation support to labels, hints and error messages.

### PR DESCRIPTION
**Pull request by martndemus**
Fixes #64

### CODE REVIEW

**Comment by martndemus** [comment]
cc @lcpriest, can you work with this?

**Comment by lcpriest** [comment]
This is great! I went one step further in my fork, with 'modelName.placeholders.email', 'modelName.attributes.email' (I should probably have used 'modelName.labels.email') and 'modelName.hints.email'. 

Thoughts?

File: addon/components/form-field.js [hunk 1]
```diff
@@ -60,15 +62,32 @@ const FormFieldComponent = Component.extend({
...
+  }),
+
+  modelName: or('object.modelName', 'object.constructor.modelName'),
+
+  labelI18nKey: computed('propertyName', function() {
+    return [
+      get(this, 'i18nKeyPrefix'),
+      get(this, 'modelName'),
+      get(this, 'propertyName')
+    ].filter((x) => !!x)
```
Comment (lcpriest) [inline addon/components/form-field.js]: would `compact().join('.')` work too?

File: addon/components/form-field.js [hunk 2]
```diff
@@ -60,15 +62,32 @@ const FormFieldComponent = Component.extend({
   },
 
   labelText: computed('propertyName', 'label', function() {
-    return get(this, 'label') || humanize(get(this, 'propertyName'));
+    let i18n = get(this, 'i18n');
+
+    if (isPresent(i18n)) {
+      return i18n.t(get(this, 'labelI18nKey'));
```
Comment (lcpriest) [inline addon/components/form-field.js]: I also did a check for the existence of the key, before falling back to the normal flow.

`i18n.exists(key)` means that you can still pass in the label normally (or not at all).

**Comment by lcpriest** [comment]
For context, I was trying to mimic https://github.com/plataformatec/simple_form#i18n

**Pull request merged**